trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py

```diff
@@ -173,39 +170,25 @@ def compute_aero_derivatives(segment):
-    # Flap deflection perturbation
-    
-    perturbed_segment = deepcopy(segment)    
-    delta_plus        = delta + 0.1
-    perturbed_segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.flap.deflection = delta_plus
-    
-    iterate = perturbed_segment.process.iterate
-    iterate.conditions(perturbed_segment) 
-    
-    # set segment derivatives based on perturbed segment
-    dDelta_Flap     = perturbed_segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.flap.deflection - segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.flap.deflection
-    dCL             = perturbed_segment.state.conditions.aerodynamics.lift_coefficient - segment.state.conditions.aerodynamics.lift_coefficient
-    dCL_dDelta_Flap = dCL/dDelta_Flap
-    
-    segment.state.conditions.aero_derivatives.dCL_dDelta_Flap = dCL_dDelta_Flap
-    
-    # ----------------------------------------------------------------------------    
-    # Slat deflection perturbation
-    
-    perturbed_segment = deepcopy(segment)    
-    delta_plus        = delta + 0.1
-    perturbed_segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.slat.deflection = delta_plus
-    
-    iterate = perturbed_segment.process.iterate
-    iterate.conditions(perturbed_segment) 
-    
-    # set segment derivatives based on perturbed segment
-    dDelta_Slat     = perturbed_segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.flap.deflection - segment.analyses.aerodynamics.geometry.wings.main_wing.control_surfaces.flap.deflection
-    dCL             = perturbed_segment.state.conditions.aerodynamics.lift_coefficient - segment.state.conditions.aerodynamics.lift_coefficient
-    dCL_dDelta_Slat = dCL/dDelta_Slat
-    
-    segment.state.conditions.aero_derivatives.dCL_dDelta_Slat = dCL_dDelta_Slat    
-    
+    # Control surface deflection perturbation for each wing
+    for wing in list(segment.analyses.aerodynamics.geometry.wings.keys()):
+        if len(segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces) !=0:
+            # set segment derivatives based on perturbed segment
+            for cs in list(wing.control_surfaces):
+                delta             = segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection                
+                perturbed_segment = deepcopy(segment)    
+                delta_plus        = delta + 0.1
+                perturbed_segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection = delta_plus
+                
+                iterate = perturbed_segment.process.iterate
+                iterate.conditions(perturbed_segment)                 
+                
+                dDelta          = perturbed_segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection - segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection
```

Comment:
This is much better!

However, can you also calculate the other aero derivatives Cl, Cn, Cm, etc..? That would so useful!

Reply:
Yes - added the others as well